crates/core/src/view/search_bar.rs

```diff
@@ -20,55 +20,77 @@ pub struct SearchBar {
 }
 
 impl SearchBar {
-    pub fn new(rect: Rectangle, input_id: ViewId, placeholder: &str, text: &str, context: &mut Context) -> SearchBar {
-        let id = ID_FEEDER.next();
-        let mut children = Vec::new();
+    pub fn new_empty(availible_space: Rectangle) -> SearchBar {
         let dpi = CURRENT_DEVICE.dpi;
+        let small_height = scale_by_dpi(SMALL_BAR_HEIGHT, dpi) as i32;
         let thickness = scale_by_dpi(THICKNESS_MEDIUM, dpi) as i32;
-        let side = rect.height() as i32;
+        let bar_height = small_height + thickness;
+
+        SearchBar {
+            id: ID_FEEDER.next(),
+            children: Vec::new(),
+            // NOTE - SearchBar is a bottom-aligned component
+            rect: rect![
+                availible_space.min.x, availible_space.max.y - bar_height,
+                availible_space.max.x, availible_space.max.y]
+        }
+    }
+    pub fn new(availible_space: Rectangle, input_id: ViewId, placeholder: &str) -> SearchBar {
```

Comment:
Did not break the sub-components out into their own methods in this component, because I think we should rethink how the search bar works (group it into a single pop up with the keyboard, and allow for custom event handling, like the icon buttons do)